cli/src/cmd_net.rs

```diff
@@ -0,0 +1,1485 @@
+            .send()
+            .await?
+            .into_inner()
+            .into_iter()
+            .map(|x| BgpAnnouncementCreate {
+                address_lot_block: NameOrId::Id(x.address_lot_block_id),
+                network: x.network,
+            })
+            .collect();
+
+        current.push(BgpAnnouncementCreate {
+            address_lot_block: NameOrId::Name(self.address_lot.clone()),
+            network: self.prefix.to_string().parse().unwrap(),
+        });
+
+        ctx.client()?
+            .networking_bgp_announce_set_update()
+            .body(BgpAnnounceSetCreate {
+                announcement: current,
+                name: self.announce_set.clone(),
+                description: self.announce_set.to_string(), //TODO?
+            })
+            .send()
+            .await?;
+
+        Ok(())
+    }
+}
+
+#[derive(Parser, Debug, Clone)]
+#[command(verbatim_doc_comment)]
+#[command(name = "announce")]
+pub struct CmdBgpWithdraw {
+    /// The announce set to withdraw from.
+    announce_set: Name,
+
+    /// The prefix to withdraw.
+    prefix: oxnet::IpNet,
+}
+
+#[async_trait]
+impl RunnableCmd for CmdBgpWithdraw {
+    async fn run(&self, ctx: &Context) -> Result<()> {
+        let mut current: Vec<BgpAnnouncementCreate> = ctx
+            .client()?
+            .networking_bgp_announce_set_list()
+            .name_or_id(NameOrId::Name(self.announce_set.clone()))
+            .send()
+            .await?
+            .into_inner()
+            .into_iter()
+            .map(|x| BgpAnnouncementCreate {
+                address_lot_block: NameOrId::Id(x.address_lot_block_id),
+                network: x.network,
+            })
+            .collect();
+
+        current.retain(|x| x.network.to_string() != self.prefix.to_string());
+
+        ctx.client()?
+            .networking_bgp_announce_set_update()
+            .body(BgpAnnounceSetCreate {
+                announcement: current,
+                name: self.announce_set.clone(),
+                description: self.announce_set.to_string(), //TODO?
+            })
+            .send()
+            .await?;
+
+        Ok(())
+    }
+}
+
+#[derive(clap::ValueEnum, Clone, Debug)]
+enum FilterDirection {
+    Import,
+    Export,
+}
+
+#[derive(Parser, Debug, Clone)]
+#[command(verbatim_doc_comment)]
+#[command(name = "announce")]
+pub struct CmdBgpFilter {
+    /// Id of the rack to add the address to.
+    rack: Uuid,
+
+    /// Switch to add the address to.
+    #[arg(value_enum)]
+    switch: Switch,
+
+    /// Port to add the port to.
+    #[arg(value_enum)]
+    port: Port,
+
+    /// Peer to apply allow list to.
+    peer: IpAddr,
+
+    /// Whether to apply the filter to imported or exported prefixes.
+    #[arg(value_enum)]
+    direction: FilterDirection,
+
+    /// Prefixes to allow for the peer.
+    allowed: Vec<oxnet::IpNet>,
+
+    /// Do not filter. Takes precedence over allowed list.
```

Comment:
you can use a `clap::Group` for mutual exclusion although I'm not sure how that works with `allowed` being a `Vec`

Reply:
I used `conflicts_with` here.